### DISCUSSION TITLE
vala language server with gtk completion

### DISCUSSION THREAD

**Discussion by Ok9xNirab** [Prince781/vala-language-server#201] [Q&A] [answered]
Hi,
I'm newbie person to learn vala and gtk development . 
I got some errors from vs-code . 

But code is compiled and run smoothly . no problem !!
why i seen this wrong ERRORS ??

<img src="https://i.ibb.co/HdTgTBL/Screenshot-from-2021-07-03-23-43-26.png" />

--

<img src="https://i.ibb.co/YRxfvgT/Screenshot-from-2021-07-03-23-43-49.png" />

Also got errors from this line of code :

```

Gtk.Window()
public GLib.Settings settings;
public Gtk.Stack stack { get; set; }

```

**Comment by Prince781** [comment, marked as answer]
@Ok9xNirab a few questions:

1. What is the version of VLS? How did you install it?
2. What is the version of Vala? Run `valac --version` and `valac --api-version`
3. What is the version of Meson? Run `meson --version`
3. Are there any messages shown in VSCode like "failed to configure Meson project"? From the second image it looks like the `gtk` package was not picked up, which suggests that Meson configuration failed.

**Reply by Ok9xNirab** [reply]
Installed VLS via [obs](https://software.opensuse.org//download.html?project=home%3APrince781&package=vala-language-server) . i use popOS , which is ubuntu 21.04. i installed VLS by .deb package installer .

<img src="https://i.ibb.co/v1Z6KDQ/Screenshot-from-2021-07-04-00-43-13.png" />


vala version : `v0.48.17`
meson version: `v0.56.2`

**Reply by Prince781** [reply]
Okay, this all looks good. Now when you do `meson build && meson compile -C build` in terminal does it work?

**Reply by Ok9xNirab** [reply]
output is :

```
Directory already configured.

Just run your build command (e.g. ninja) and Meson will regenerate as necessary.
If ninja fails, run "ninja reconfigure" or "meson --reconfigure"
to force Meson to regenerate.

If build failures persist, run "meson setup --wipe" to rebuild from scratch
using the same options as passed when configuring the build.
To change option values, run "meson configure" instead.
Found runner: ['/usr/bin/ninja']
ninja: Entering directory `build'
[6/6] Linking target src/com.github.alecaddd.jarvis
```

**Reply by Prince781** [reply]
What do you see in VSCode status? If you enable debug mode for the Vala plugin, restart, and then open the output window in VSCode to "Vala Language Server", there should be a lot of messages. Copy and paste them here.

**Reply by Ok9xNirab** [reply]
Oops !! My issue has been fixed . Don't know , How ?? have you updated extension ??

**Reply by Prince781** [reply]
I didn't update the extension. I guess it's good that your issue was fixed. Do you have any idea why?

**Reply by Ok9xNirab** [reply]
I think some assets need download after VLS install !!
